src/publisher/services/publisher-api.ts

```diff
@@ -437,14 +438,17 @@ export class PublisherApi {
     }).then((response) => response.json() as unknown as FilterTable[]);
   }
 
-  public async assignSources(datasetId: string, sourceTypeAssignment: SourceAssignmentDTO[]): Promise<DatasetDTO> {
+  public async assignSources(
+    datasetId: string,
+    sourceTypeAssignment: SourceAssignmentDTO[]
+  ): Promise<{ dataset: DatasetDTO; build_id: string }> {
```

Comment:
Can we have a new interface for this return type please - maybe `DatasetBuildDTO`?